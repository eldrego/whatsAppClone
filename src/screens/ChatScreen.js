import React, { Component } from 'react';
import {
  View,
  Text,
  ImageBackground,
  FlatList,
} from 'react-native';
import api from '../services/api';
import styles from '../lib/css/styles';

export default class Home extends Component {
  static navigationOptions = ({ navigation }) => ({
    title: `Chat with ${navigation.state.params.name}`,
  })

  constructor(props) {
    super(props);

    this.state = {
      messages: [],
    };
  }

  componentDidMount() {
    api.getMockData().then((messages) => {
      this.setState({
        messages,
      });
    });
  }

  render() {
    return (
      <ImageBackground
        style={styles.container}
        source={require('../images/bg.png')}
      >
        <View>
          <FlatList
            data={this.state.messages}
            renderItem={({ item }) =>
              <View><Text>{item.message}</Text></View>
            }
            keyExtractor={(item, index) => (`message-${index}`)}
          />
        </View>
      </ImageBackground>
    );
  }
}
