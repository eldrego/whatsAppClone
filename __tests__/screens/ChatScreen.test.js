import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ChatScreen from '../../src/screens/ChatScreen';

const navigate = jest.fn();
const goBack = jest.fn();
configure({ adapter: new Adapter() });

it('renders correctly', () => {
  const tree = renderer.create(<ChatScreen navigation={{ navigate }} />);
  expect(tree).toMatchSnapshot();
});

xit('navigate on press', () => {
  const chatScreen = shallow(<ChatScreen navigation={{ navigate }} />);
  chatScreen.find('Button').simulate('press');
  expect(goBack).toBeCalledWith('home');
});

it('has proper navigation options', () => {
  expect(ChatScreen.navigationOptions({
    navigation: {
      state: {
        params: {
          name: 'John',
        },
      },
    },
  })).toMatchSnapshot();
});
