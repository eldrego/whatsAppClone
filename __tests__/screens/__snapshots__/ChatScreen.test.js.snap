// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has proper navigation options 1`] = `
Object {
  "title": "Chat with John",
}
`;

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 20,
    }
  }
>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;
